browser/scripts/application.js

```diff
@@ -629,6 +630,8 @@ Application.prototype.clearSelection = function() {
 	this.selectedNodes = [];
 	this.selectedConnections = [];
 
+	if (E2.ui) E2.ui.state.selectedObjects = this.selectedNodes
```

Comment:
don't need these constant checks. let's assume we always have E2.ui.state, if not, we can inject a mock